lib/fetch-libc++.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+GCC_VERSION=$1
+
+if [ $GCC_VERSION = "12.2.0" ]; then
+  GCC_SHA="f879b8763de38b34ab904d1ed8733e9accf60f4f9ccab78462bb525767691420"
+fi
+
+# Name of the pre-created compiled directories.
+ZIP_FILE=libtock-libc++-$GCC_VERSION.zip
+
+# List of mirrors we support.
+MIRRORS=(\
+  "https://www.cs.virginia.edu/~bjc8c/archive/tock"\
+  "https://alpha.mirror.svc.schuermann.io/files"\
+)
```

Comment:
I don't think we should be automatically downloading binaries from custom places.

What happens if these are replaced with something malicious?
What happens if this is updated or changed, it isn't tracked in git?

Reply:
The sha256 hash is. If someone can replace the zip with something malicious and the same hash, maybe we accept defeat?

Reply:
What if we bump the newlib version and then delete the old one? That will break users who don't update as it's not checked into git.

It just seems risky and clunky to have these outside the repo. What is the benefit?

Reply:
Hopefully we wouldn't delete from every mirror? I don't know why we would delete them.

The benefit is not having to not having to add ~150MB of zip files to the repo on every version update.

Reply:
> Hopefully we wouldn't delete from every mirror?

Hopefully not, but nothing stops that from happening.

> The benefit is not having to not having to add ~150MB of zip files to the repo on every version update.

With git-lfs is that really an issue?

Reply:
This all comes from a discussion we had where I thought the conclusion was pulling in a zip (like we've done with the riscv toolchain) was fine. https://github.com/tock/tock/blob/master/doc/wg/core/notes/core-notes-2023-09-22.md#libtock-c-newlib-updates

@lschuermann @ppannuto

Reply:
After another discussion I don't think we are eager to pay for github lfs when we have a decent (free, ready to go) solution. https://docs.github.com/en/billing/managing-billing-for-git-large-file-storage/about-billing-for-git-large-file-storage

Reply:
Happy to continue hosting this on my personal mirror. I can also add a mirror on our cluster web server at Princeton. I believe that, as long as we have a system that checks these URLs and complains if they're down, we should be able to maintain a good number of toolchain versions reliably before bandwidth / storage becomes an issue for us.

Reply:
FWIW here's a crude example of how we can use GitHub actions to periodically check whether our mirrors are still alive and the links still work: https://github.com/lschuermann/tock-mirrortest. We can add a step the workflow that automatically opens an issue and tags the admins of the failing mirror, and potentially also a script that collects these links from the various Tock repos. Just need to make sure we don't over-engineer this too much. What do you think @bradjc @alevy?

Reply:
> After another discussion I don't think we are eager to pay for github lfs when we have a decent (free, ready to go) solution. https://docs.github.com/en/billing/managing-billing-for-git-large-file-storage/about-billing-for-git-large-file-storage

Ah, I didn't realise GitHub charges for LFS